### PR TITLE
[Scatter2] APIs, provenance, shorter URL

### PR DESCRIPTION
**Pull request by intrepiditee**
1. Use `/stat/collection` for getting date and `/place/stat-vars/union` for getting available statvars
2. Show provenance #702 
3. Create shorter URL hash
4. Clear hash after resetting #704

### CODE REVIEW

File: static/js/tools/scatter2/plot_options.tsx [hunk 1]
```diff
@@ -154,6 +201,51 @@ function PlotOptions(): JSX.Element {
   );
 }
 
+function getCurrentYear(): number {
+  return new Date().getFullYear();
+}
+
+function getNToM(n: number, m: number, descending = false): Array<number> {
```
Comment (beets) [inline static/js/tools/scatter2/plot_options.tsx:208]: please add brief descriptions for these helpers

File: static/js/tools/scatter2/statvar.tsx [hunk 2]
```diff
@@ -199,40 +181,88 @@ function StatVarChooser(): JSX.Element {
...
+  x: AxisWrapper,
+  y: AxisWrapper,
+  statVars: Set<string>
+) {
+  let message = "";
+  const statVarX = _.findKey(x.value.statVar);
+  const statVarY = _.findKey(y.value.statVar);
+  if (statVarX && !statVars.has(statVarX)) {
+    x.unsetStatVar();
+    message += `Sorry, no data available for ${statVarX}`;
```
Comment (beets) [inline static/js/tools/scatter2/statvar.tsx:255]: Can we provide an action to the user here - try another stat var or change the constraints.

**Review by beets — COMMENTED** [review]
Mostly looks good! Just a few comments.

File: static/js/tools/scatter2/statvar.tsx (hunk 2 above)
Comment (intrepiditee) [reply static/js/tools/scatter2/statvar.tsx:259]: fixed

File: static/js/tools/scatter2/plot_options.tsx (hunk 1 above)
Comment (intrepiditee) [reply static/js/tools/scatter2/plot_options.tsx:216]: fixed

File: server/routes/api/place.py [hunk 3]
```diff
@@ -162,6 +160,40 @@ def statsvars(dcid):
...
+    """Get all the statistical variable dcids for some places.
+
+    Args:
+        dcids: Place DCIDs separated by "^" as a single string.
+
+    Returns:
+        List of unique statistical variable dcids each as a string.
+    """
+    dcids = dcids.split("^")
+    # The two indexings are due to how protobuf fields are converted to json
```
Comment (beets) [inline server/routes/api/place.py:174]: That's unfortunate... we should hide these complexities from the user. We should be able to remove that outer statVars. Please add a todo to update this.

**Review by beets — APPROVED** [review]
thanks for the fixes! we should hide the implementation detail to avoid the double indexing of the union api, but that's a mixer fix. i'll merge this in so we can test on staging.

**Pull request merged**